common/ManagementTestShared/MgmtRecordTestBase.cs

```diff
@@ -0,0 +1,224 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+#if RESOURCES_RP
+using Azure.ResourceManager.Resources;
+#else
+using Azure.Management.Resources;
+#endif
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using NUnit.Framework;
+
+
+namespace Azure.ResourceManager.TestFramework
+{
+    [Category("Recorded")]
+    public abstract class MgmtRecordTestBase<TEnvironment> : ClientTestBase where TEnvironment : TestEnvironment, new()
+    {
+        protected RecordedTestSanitizer Sanitizer { get; set; }
+
+        protected RecordMatcher Matcher { get; set; }
+
+        public TestRecording Recording { get; private set; }
+
+        public RecordedTestMode Mode { get; set; }
+
+        public TEnvironment TestEnvironment { get; }
+
+        private ResourceGroupCleanupPolicy CleanupPolicy { get; set; }
+
+        // copied the Windows version https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs
+        // as it is the most restrictive of all platforms
+        private static readonly HashSet<char> s_invalidChars = new HashSet<char>(new char[]
```

Comment:
Lets move this to ClientTestBase and make it protected since its used by RecordedTestBase as well